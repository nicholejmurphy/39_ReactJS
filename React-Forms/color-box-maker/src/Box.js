import React from "react";
import "./static/Box.css";

function Box({ id, color, width, height, removeBox }) {
  function removeMe() {
    removeBox(id);
  }

  return (
    <div className="Box">
      <div
        className="Box-container"
        style={{
          backgroundColor: color,
          width: `${width}px`,
          height: `${height}px`,
        }}
      ></div>
      <button onClick={removeMe} className="Box-remove" data-testid={id}>
        X
      </button>
    </div>
  );
}

export default Box;
