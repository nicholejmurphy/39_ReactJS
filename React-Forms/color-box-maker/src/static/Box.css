.Box {
  margin: 10px;
  text-align: center;
}

.Box-remove {
  margin-top: 5px;
}
