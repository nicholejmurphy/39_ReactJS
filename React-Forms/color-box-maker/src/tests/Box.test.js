import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "../Box";
import { v4 as uuid } from "uuid";

test("it renders without crashing", () => {
  render(<Box />);
});

test("it should match snapshot", () => {
  const { asFragment } = render(<Box />);
  expect(asFragment).toMatchSnapshot();
});
