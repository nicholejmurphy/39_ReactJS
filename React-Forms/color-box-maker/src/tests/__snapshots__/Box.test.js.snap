// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot 1`] = `[Function]`;
